integration-tests/src/test/java/com/datastax/oss/driver/api/core/cql/ExecutionInfoWarningsIT.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.datastax.oss.driver.api.core.cql;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.after;
+import static org.mockito.Mockito.verify;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.Logger;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.Appender;
+import com.datastax.oss.driver.api.core.CqlSession;
+import com.datastax.oss.driver.api.core.config.DefaultDriverOption;
+import com.datastax.oss.driver.api.testinfra.ccm.CcmRule;
+import com.datastax.oss.driver.api.testinfra.session.SessionRule;
+import com.datastax.oss.driver.api.testinfra.session.SessionUtils;
+import com.datastax.oss.driver.categories.ParallelizableTests;
+import com.datastax.oss.driver.internal.core.config.typesafe.DefaultDriverConfigLoaderBuilder;
+import com.datastax.oss.driver.internal.core.cql.CqlRequestHandler;
+import com.google.common.base.Strings;
+import java.util.List;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TestRule;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+import org.slf4j.LoggerFactory;
+
+@Category(ParallelizableTests.class)
+public class ExecutionInfoWarningsIT {
+
+  private static final int MAX_QUERY_LENGTH = 500;
+  private static final CcmRule CCM = CcmRule.getInstance();
+  private static final SessionRule<CqlSession> SESSION_RULE =
```

Comment:
which version of DSE is picked for that test if we are not specifying it explicitly?

Reply:
Tests don't pick a version, they only define version requirements (`@CassandraRequirement`, `@DseRequirement`). We pick a version when we run the test suite, with the `ccm.version` system property; if a test has requirements that don't match this version, it doesn't run.

If there is a `CassandraRequirement` and we run with DSE, we pick compare using the closest matching Cassandra version. See `CcmBridge.getCassandraVersion()`. You can also take a look at `BaseCcmRule.apply()` for more details.

In this case the test has no requirements, so it will run with whichever DSE version you start it with.